include/boost/url/any_allocator.hpp

```diff
@@ -0,0 +1,127 @@
+//
+// Created by alandefreitas on 12/01/2022.
+//
+
+#ifndef BOOST_URL_ANY_ALLOCATOR_HPP
+#define BOOST_URL_ANY_ALLOCATOR_HPP
+
+#include <cstddef>
+#include <type_traits>
+
+namespace boost {
+namespace urls {
+
+/** An allocator wrapper
+
+    This allocator erases the type of the underlying allocator
+
+ */
+template <class T>
+class any_allocator {
```

Comment:
I think you can just document it this way:
```
/** A type-erased allocator with shared ownership.
*/
#ifdef BOOST_URL_DOCS
template< class T >
using any_allocator = __see_below __;
#else
template <class T>
class any_allocator
{
  ...
#endif
```

Reply:
This wasn't resolved

Reply:
Sorry. I missed this one for some reason.

Reply:
So isn't `any_allocator` going to be public now because expert users might need to use it?

Reply:
who said it isn't going to be public?

Reply:
Implied in

```
#ifdef BOOST_URL_DOCS
template< class T >
using any_allocator = __see_below __;
#else
```

Reply:
That's still public

Reply:
But why hide the API from users in the reference?

Reply:
We aren't "hiding the API", see https://github.com/alandefreitas/url/pull/4/files#r784200210

Reply:
What benefit is there to documenting each function which is already perfectly documented in the Allocator requirements?

Reply:
So anything that satisfies a named requirement shouldn't be documented?

https://en.cppreference.com/w/cpp/memory/polymorphic_allocator